pkg/nvml/device.go

```diff
@@ -1848,6 +1970,23 @@ func (Device Device) GetGpuInstanceProfileInfo(Profile int) (GpuInstanceProfileI
 	return DeviceGetGpuInstanceProfileInfo(Device, Profile)
 }
 
+// nvml.DeviceGetGpuInstanceProfileInfoV()
+func DeviceGetGpuInstanceProfileInfoV(Device Device, Profile int, InfoV GpuInstanceProfileInfoV) Return {
```

Comment:
Question: Why can we not return the `GpuInstanceProfile` value here instead? (Like we do for the memory query).

Reply:
If you look in nvml.h, in contrast to the Memory API (which has been versioned with a specific `_v2` suffix) this API is designed to be "future proof" with a generic suffix of `V`. That means that its basic signature won't change and apps compiled against older versions of NVML will continue to function properly on future versions of NVML. This will work because v2 (and all future versions) of the `GpuInstanceProfileInfo` struct will have a `version` field as its first field, and the API requires the caller to set this `version` before passing it to the function. This allows the underlying implementation to know which version of the struct is being passed in (e.g. a `v2` struct if compiled against an older version of NVML or a `v3` struct if compiled against a newer one).

This design is fine in the C world, but it causes problems in the Go world if we want some way for the user to (1) express their intent of which struct version they want, and (2) return structs of different types based on that intent. The best we can do is have the user pass a "number" for the desired struct version and then return an interface. But then it is up to the user to extract the dynamic type of this interface before they can use it....

The compromise I came to was to make the API look similar to the C API and have the struct populated after being passed as an input rather than a return value. If we are going to be in the business of having the user pass something to indicate what version of the struct they want to use, they might as well just pass the struct itself instead of passing a number and then having to extract the dynamic type from the value returned.

An alternative I considered was to just not bother mimicking this "generic" `V` style interface and instead just go with a `_v2` style API (like was done for memory), resulting in `DeviceGetGpuInstanceProfileInfo_v2()` as the function name. This would avoid this issue because the "intent" would be encoded in the function name instead of as a parameter and we would know exactly which type to return. I was worried this might clash with a name they introduce in the future though (since that is now versioning the API, not the struct -- and they may wish to version them separately if they, for example, add a parameter to the function at some point in the future).

I'm open to more ideas before we finalize this, but this was the "best" compromise I could come up with as I was playing with it.

Reply:
What about doing this (this is tested and works):
```
// nvml.GpuInstanceGetComputeInstanceProfileInfoV()
type ComputeInstanceProfileInfoV struct{
    gpuInstance GpuInstance
    profile int
    engProfile int
}

func GpuInstanceGetComputeInstanceProfileInfoV(GpuInstance GpuInstance, Profile int, EngProfile int) ComputeInstanceProfileInfoV {
    return ComputeInstanceProfileInfoV{
        GpuInstance,
        Profile,
        EngProfile,
    }
}

func (GpuInstance GpuInstance) GetComputeInstanceProfileInfoV(Profile int, EngProfile int) ComputeInstanceProfileInfoV {
    return GpuInstanceGetComputeInstanceProfileInfoV(GpuInstance, Profile, EngProfile)
}

func (InfoV ComputeInstanceProfileInfoV) V2() (ComputeInstanceProfileInfo_v2, Return) {
    var Info ComputeInstanceProfileInfo_v2
    Info.SetVersion()
    ret := nvmlGpuInstanceGetComputeInstanceProfileInfoV(InfoV.gpuInstance, uint32(InfoV.profile), uint32(InfoV.engProfile), &Info)
    return Info, ret
}
```

With the previous iteration the caller would need to do the following:
```
var Info ComputeInstanceProfileInfo_v2
ret := GpuInstanceGetComputeInstanceProfileInfoV(GpuInstance, Profile, EngProfile, &Info)
if ret != nvml.SUCCESS {
    ...
}
```

With this new proposed way, this would turn into:
```
Info, ret := GpuInstanceGetComputeInstanceProfileInfoV(GpuInstance, Profile, EngProfile.V2()
if ret != nvml.SUCCESS {
    ...
}
```
Which is more similar to the way we typically do things (in terms of code flow), but the caller must be careful to make this extra `V2()` call at the end (or `V3()`, `V4()`, etc. in the future).

Reply:
I like this more. The question is whether there is a way to have a "default" for this struct to avoid requiring that the user explicitly select a version?

Reply:
Alternatively, we could break completely from having a `V` variant of this function and always just return the latest version (by attempting to get the latest -- and if that fails with an INVALID_ARGUMENT, go to the next previous version, and so on).

Reply:
I don't know how to provide a `default` (or even implement my last suggestion) without returning an interface type though (because any such call would need to return one of multiple types depending on what the platform supports -- forcing the user to do a dynamic type assertion anyway).

Reply:
I think I will go with what I have proposed here -- and even extend it to include `V1()` so that users can begin to "standardize" on using the `V` API for everything. I actually think it's fine that they pick a version and stick with it because the version they choose implies the minimum driver version their code will run on.

Reply:
Updated